pkg/controller/install/certresources.go

```diff
@@ -459,32 +465,33 @@ func (i *StrategyDeploymentInstaller) installCertRequirementsForDeployment(deplo
 	authDelegatorClusterRoleBinding.SetName(AuthDelegatorClusterRoleBindingName(service.GetName()))
 	authDelegatorClusterRoleBinding.SetLabels(map[string]string{OLMManagedLabelKey: OLMManagedLabelValue})
 
-	existingAuthDelegatorClusterRoleBinding, err := i.strategyClient.GetOpLister().RbacV1().ClusterRoleBindingLister().Get(authDelegatorClusterRoleBinding.GetName())
-	if err == nil {
+	// Optimistically create the ClusterRoleBinding with CSV as the labelled owner
+	if err := ownerutil.AddOwnerLabels(authDelegatorClusterRoleBinding, i.owner); err != nil {
+		return nil, nil, err
+	}
+	_, err = i.strategyClient.GetOpClient().CreateClusterRoleBinding(authDelegatorClusterRoleBinding)
+	if apierrors.IsAlreadyExists(err) {
+		// ClusterRoleBinding already exists; update if safe to do so.
+		existingAuthDelegatorClusterRoleBinding, err := i.strategyClient.GetOpLister().RbacV1().ClusterRoleBindingLister().Get(authDelegatorClusterRoleBinding.GetName())
+		if err != nil {
+			return nil, nil, err
```

Comment:
I like the more descriptive error given on https://github.com/operator-framework/operator-lifecycle-manager/pull/3159/files#diff-bc7b779225ba70ff0e1f63ff55cb2dd3d71b5ae4f96c158947266da9f8fde60dR262 - could that be done here as well?

Reply:
Absolutely!

Reply:
Updated, thanks!